app/main/lib/helpers.py

```diff
@@ -1,9 +1,22 @@
 def context_matches(query_context, item_context):
+  """
+    Check a pair of contexts to determine if they match - first pass is
+    typically overly permissive and based in postgres - this test does an
+    exact check. We do *not* check for equality or present for project_media_id,
+    as they are unique to all items, and as a result, if we did use it as a
+    comparator when looking at the contexts, we would never match any items, ever.
+    Add more globally unique context ids into that list. It's a hack for an
+    assumption we try to make everywhere else that we are agnostic about
+    context values, but in this small situation, we actually need to check
+    against this.
+  """
+  irrelevant_keys = ["project_media_id"]
```

Comment:
I'm curious what lead to this change. As far as I can tell, `context_matches` is only used in the `search_by_context` functions and these are not used anywhere in Alegre itself (aside from test functions). 
https://github.com/search?q=repo%3Ameedan%2Falegre%20search_by_context&type=code

IMHO, it should be up to the caller of Alegre's API to remove any "irrelevant_keys" from the query context. I.e., Alegre should not have this knowledge and if Check API needs to search by context it should be Check API---not Alegre---that removes the `project_media_id`

I'm wondering, however, if I'm missing something where we're using this function in a new /different way with the refactor? I just cannot see where :pray: